feat: spartan/bootstrap.sh network_deploy

just-mitch opened the pull request:
Drop the effort to run local workflows via `act`​ and instead adopt the existing pattern of:

1. configure in github action
2. call bootstrap

You can now, e.g.:

```
./spartan/bootstrap.sh network_deploy scenario.internal.env
./spartan/bootstrap.sh single_test scenario.local.env spartan/smoke.test.ts
```

- Add documentation in [DEVELOP.md](http://DEVELOP.md)
- Fix deployments after changes to json processing
- nuke a bunch of unused github actions
- write github action to do scenario network creation/tests and staging-public network creation

just-mitch commented:
* **#16639** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16639?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>: 2  dependent PRs ([#16648](https://github.com/AztecProtocol/aztec-packages/pull/16648) <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16648?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>, [#16681](https://github.com/AztecProtocol/aztec-packages/pull/16681) <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16681?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>) 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16639?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#16595** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16595?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#16580** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16580?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#16553** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16553?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#16549** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16549?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#16538** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16538?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

just-mitch commented on spartan/scripts/install_deps.sh:
This was persistently asking me for `sudo` when I was running locally.

just-mitch committed:
feat: spartan/bootstrap.sh network_deploy

just-mitch commented on .github/workflows/deploy-staging-networks.yml:
```diff
@@ -0,0 +1,83 @@
+          NAMESPACE="v${{ github.event.client_payload.major_version }}-staging-public"
+          AZTEC_DOCKER_IMAGE="aztecprotocol/aztec:${{ github.event.client_payload.semver }}"
+          ETHEREUM_CHAIN_ID=11155111
+          ETHEREUM_RPC_URLS=${{ secrets.SEPOLIA_RPC_URLS }}
```
These secrets need to be set.

spalladino commented on yarn-project/end-to-end/src/spartan/DEVELOP.md:
````diff
@@ -0,0 +1,116 @@
+
+```bash
+# run one
+./spartan/bootstrap.sh test_single scenario.local.env spartan/smoke.test.ts
````
```suggestion
./spartan/bootstrap.sh single_test scenario.local.env spartan/smoke.test.ts
```

spalladino commented on yarn-project/end-to-end/src/spartan/smoke.test.ts:
```diff
@@ -23,45 +24,54 @@ describe('smoke test', () => {
+          return committee !== undefined;
+        },
+        'committee',
+        60 * 60, // wait up to 1 hour, since if the rollup was just deployed there will be no committee for 2 epochs
```
We should parameterize timeouts based on the rollup settings

spalladino commented on yarn-project/ethereum/src/deploy_l1_contracts.ts:
```diff
@@ -854,7 +854,7 @@ export const addMultipleValidators = async (
         },
         {
-          gasLimit: 45_000_000n,
+          gasLimit: 32_000_000n,
```
Why this change?

just-mitch replied:
Ah! Because I didn't realize the block gas limit was raised to 45M!

spalladino commented on spartan/aztec-node/values.yaml:
```diff
@@ -114,6 +114,24 @@ node:
+
+  sentinel:
+    enabled: true
+  slash:
+    minPenaltyPercentage: 0.5
+    maxPenaltyPercentage: 2.0
+    inactivityTargetPercentage: 0.7
+    inactivityPenalty: 0.0
+    prunePenalty: 0.0
+    dataWithholdingPenalty: 0.0
+    proposeInvalidAttestationsPenalty: 0.0
+    attestDescendantOfInvalidPenalty: 0.0
+    unknownPenalty: 0.0
+    invalidBlockPenalty: 0.0
+    offenseExpirationRounds: 8
+    maxPayloadSize: 50
```
I had set all slash to empty values, so we'd use the defaults defined in the node and we don't have multiple defaults floating around. Any reason for bringing them back?

https://github.com/AztecProtocol/aztec-packages/blob/395632f31add5159f0574ae17494efab2d6ea19e/spartan/aztec-network/values.yaml#L111-L131

just-mitch replied:
Good call. Thanks.

spalladino commented on spartan/scripts/deploy_network.sh:
```diff
@@ -0,0 +1,292 @@
+AZTEC_SLASH_AMOUNT_SMALL=${AZTEC_SLASH_AMOUNT_SMALL:-10000000000000000000}
+AZTEC_SLASH_AMOUNT_MEDIUM=${AZTEC_SLASH_AMOUNT_MEDIUM:-20000000000000000000}
+AZTEC_SLASH_AMOUNT_LARGE=${AZTEC_SLASH_AMOUNT_LARGE:-50000000000000000000}
+AZTEC_SLASHER_FLAVOR=${AZTEC_SLASHER_FLAVOR:-tally}
```
Same comment as above on duplicate defaults

spalladino commented on spartan/scripts/deploy_network.sh:
```diff
@@ -0,0 +1,292 @@
+JOB_NAME = "deploy-rollup-contracts"
+JOB_BACKOFF_LIMIT = 3
+JOB_TTL_SECONDS_AFTER_FINISHED = 3600
+EOF
```
Should we instead dump all aztec-related env vars here? We should be able to source the list of all vars from `env-vars.ts`.

just-mitch replied:
Hm. I think we could. I guess there are a few outside env-vars.ts that we'd want to set.

spalladino commented on spartan/scripts/deploy_network.sh:
```diff
@@ -0,0 +1,292 @@
+DEPLOY_AZTEC_INFRA_DIR="${SCRIPT_DIR}/../terraform/deploy-aztec-infra"
+"${SCRIPT_DIR}/override_terraform_backend.sh" "${DEPLOY_AZTEC_INFRA_DIR}" "${CLUSTER}" "${BASE_STATE_PATH}/deploy-aztec-infra/${SALT}"
+
+cat > "${DEPLOY_AZTEC_INFRA_DIR}/terraform.tfvars" << EOF
```
Same comment as above on dumping env vars

spalladino commented on spartan/terraform/deploy-aztec-infra/variables.tf:
```diff
@@ -186,5 +174,90 @@ variable "PROVER_MNEMONIC_START_INDEX" {
+  default     = true
+}
+
+variable "SLASH_MIN_PENALTY_PERCENTAGE" {
+  description = "The slash min penalty percentage"
+  type        = string
+  default     = 0.5
+}
```
Same comment as above on not duplicating default values

just-mitch replied:
It's a great call. I'll remove all the defaults from terraform. No need to add layers here.

spalladino commented on spartan/aztec-node/templates/_pod-template.yaml:
```diff
@@ -179,6 +179,34 @@ spec:
           value: "{{ .Values.service.p2p.port }}"
+        - name: PROVER_REAL_PROOFS
+          value: "{{ .Values.node.proverRealProofs }}"
+        - name: SENTINEL_ENABLED
+          value: "{{ .Values.node.sentinel.enabled }}"
+        - name: SLASH_MIN_PENALTY_PERCENTAGE
+          value: "{{ .Values.node.slash.minPenaltyPercentage }}"
+        - name: SLASH_MAX_PENALTY_PERCENTAGE
+          value: "{{ .Values.node.slash.maxPenaltyPercentage }}"
+        - name: SLASH_INACTIVITY_TARGET_PERCENTAGE
+          value: "{{ .Values.node.slash.inactivityTargetPercentage }}"
+        - name: SLASH_INACTIVITY_PENALTY
+          value: "{{ .Values.node.slash.inactivityPenalty }}"
+        - name: SLASH_PRUNE_PENALTY
+          value: "{{ .Values.node.slash.prunePenalty }}"
+        - name: SLASH_DATA_WITHHOLDING_PENALTY
+          value: "{{ .Values.node.slash.dataWithholdingPenalty }}"
+        - name: SLASH_PROPOSE_INVALID_ATTESTATIONS_PENALTY
+          value: "{{ .Values.node.slash.proposeInvalidAttestationsPenalty }}"
+        - name: SLASH_ATTEST_DESCENDANT_OF_INVALID_PENALTY
+          value: "{{ .Values.node.slash.attestDescendantOfInvalidPenalty }}"
+        - name: SLASH_UNKNOWN_PENALTY
+          value: "{{ .Values.node.slash.unknownPenalty }}"
+        - name: SLASH_INVALID_BLOCK_PENALTY
+          value: "{{ .Values.node.slash.invalidBlockPenalty }}"
+        - name: SLASH_OFFENSE_EXPIRATION_ROUNDS
+          value: "{{ .Values.node.slash.offenseExpirationRounds }}"
+        - name: SLASH_MAX_PAYLOAD_SIZE
+          value: "{{ .Values.node.slash.maxPayloadSize }}"
```
I had injected these in the `validator.yaml` template, should we remove them from either of the two?

https://github.com/AztecProtocol/aztec-packages/blob/395632f31add5159f0574ae17494efab2d6ea19e/spartan/aztec-network/templates/validator.yaml#L194-L253

just-mitch replied:
Ah nice. I've used what you have. Very soon I think we will remove the aztec-network chart.

spalladino reviewed (approved):
Approving since this is something we definitely want, but I'm worried about the complexity of configuration. If I managed to follow the flow correctly, what happens with a config value is:

- Gets defined in an `environments/scenario.env` file
- Gets loaded into `scripts/deploy-network.sh` (which has its set of defaults)
- Gets written into a tfvars file for terraform to pick up
- Gets picked up by terraform from a `variables.tf` file (which has its set of defaults)
- Gets passed onto `custom_settings` for a helm release resource in a tf definition
- Gets loaded into helm based on a `values.yaml` file (which has its set of defaults)
- Gets injected as an env var from a template such as `validator.yaml` (or `_pod-template.yaml`?)
- Gets loaded by the node (which has its set of defaults)

Also, it seems there are now duplicate configurations for slashing in helm. I added comments to point to them.

just-mitch commented:
> Approving since this is something we definitely want, but I'm worried about the complexity of configuration. If I managed to follow the flow correctly, what happens with a config value is:
> 
> * Gets defined in an `environments/scenario.env` file
> * Gets loaded into `scripts/deploy-network.sh` (which has its set of defaults)
> * Gets written into a tfvars file for terraform to pick up
> * Gets picked up by terraform from a `variables.tf` file (which has its set of defaults)
> * Gets passed onto `custom_settings` for a helm release resource in a tf definition
> * Gets loaded into helm based on a `values.yaml` file (which has its set of defaults)
> * Gets injected as an env var from a template such as `validator.yaml` (or `_pod-template.yaml`?)
> * Gets loaded by the node (which has its set of defaults)
> 
> Also, it seems there are now duplicate configurations for slashing in helm. I added comments to point to them.

You're absolutely right in the current flow, and its unnecessary complexity. I removed the defaults out of deploy-network.sh, and the variables.tf, and the helm values yaml. Now if nothing is set, it just uses the default values in TS.

The flow you've described is still correct, but removing all the defaults makes it much easier to reason about.
